Automattic/wp-calypso
# Stats: Updating Tags module

Related to #72134, #72133

## Proposed Changes

* modernising `Tags & categories` module
* extending horizontal bar component to display an array of items for a label (resulting in a concatenated label with icons)
* refactoring the way the icons are generated for the left side of the labels
* accepting a function for the left side icon (to render it for child items)
* extending horizontal bar groups to allow placement of group chevron on either the left or right side of the label (previously only right side)
* removing obsolete value sorting for the Search terms module
* fixing label ellipsis for group labels
* if a child item doesn't have any value, apply the parent's value (grouped tags are grouped because of the value but don't have the value themselves) 

## Testing Instructions



* go to the live branch and the `Insights` page
* verify that the current `Tags & categories` renders correctly and you can expand collapsed groups
* apply feature flag to the URL `?flags=stats/insights-page-grid`
* verify that the new module with horizontal bars loads correctly and collapsing/expanding works
* check modules on the `Traffic` page for potential missing avatars or unnecessary spacing between items that haven't been there before

| Before | After |
| --- | --- |
| ![SCR-20230217-u0r](https://user-images.githubusercontent.com/112354940/219595175-2748226c-a01e-4d26-98b5-4f90b14da0a7.png) | ![SCR-20230217-tzd](https://user-images.githubusercontent.com/112354940/219595258-b8c1ff84-193e-47db-8c06-482af2295650.png) |


## Pre-merge Checklist
